src/components/DisplayCard/DisplayCard.tsx

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import {
+  Box,
+  Divider,
+  HStack,
+  StylesProvider,
+  Text,
+  useMultiStyleConfig,
+  useStyles,
+  VStack,
+} from "@chakra-ui/react"
+import type {
+  BoxProps,
+  omitThemingProps,
+  StackProps,
+  TextProps,
+} from "@chakra-ui/react"
+import { createContext } from "react"
+
+import {
+  DisplayCardVariant,
+  DISPLAY_CARD_THEME_KEY,
+} from "theme/components/DisplayCard"
+
+interface DisplayCardProps extends StackProps {
+  variant?: DisplayCardVariant
+}
+
+interface DisplayCardHeaderProps extends BoxProps {
+  button?: JSX.Element
+}
+
+interface DisplayCardTitleProps extends TextProps {
+  icon?: JSX.Element
+  hasUnderline?: boolean
+}
+
+interface DisplayCardContextProps extends ReturnType<typeof omitThemingProps> {
+  variant: DisplayCardVariant
+}
+
+const DisplayCardContext = createContext<DisplayCardContextProps | undefined>(
+  undefined
+)
+
+export const DisplayCard = ({
+  bgColor = "background.action.defaultInverse",
+  children,
+  ...props
+}: DisplayCardProps): JSX.Element => {
+  const variant = props.variant ?? "onlyHeader"
+  const styles = useMultiStyleConfig(DISPLAY_CARD_THEME_KEY, props)
+
+  return (
+    <DisplayCardContext.Provider value={{ variant }}>
+      <StylesProvider value={styles}>
+        <VStack
+          position="relative"
+          h="100%"
+          spacing="1rem"
+          bgColor={bgColor}
+          {...(styles.container as StackProps)}
+        >
+          {children}
+        </VStack>
+      </StylesProvider>
+    </DisplayCardContext.Provider>
+  )
+}
+
+export const DisplayCardHeader = ({
+  button,
+  children,
+  ...props
+}: DisplayCardHeaderProps): JSX.Element => {
+  const styles = useStyles()
+
+  return (
+    <Box __css={styles.header} {...props}>
```

Comment:
Hmm wait which one do you mean? The styles here is for the Box component, so that it take up the entire width of the card, which is required to force the underline to stretch the entire card if it exists.